packages/@aws-cdk/aws-apigateway/lib/integrations/stepfunctions.ts

```diff
@@ -0,0 +1,276 @@
+    }
+
+    let deploymentToken;
+
+    if (stateMachineName !== undefined && !Token.isUnresolved(stateMachineName)) {
+      deploymentToken = JSON.stringify({ stateMachineName });
+    }
+    return {
+      ...bindResult,
+      deploymentToken,
+    };
+  }
+}
+
+/**
+ * Defines the integration response that passes the result on success,
+ * or the error on failure, from the synchronous execution to the caller.
+ *
+ * @returns integrationResponse mapping
+ */
+function integrationResponse() {
+  const errorResponse = [
+    {
+      /**
+       * Specifies the regular expression (regex) pattern used to choose
+       * an integration response based on the response from the back end.
+       * In this case it will match all '4XX' HTTP Errors
+       */
+      selectionPattern: '4\\d{2}',
+      statusCode: '400',
+      responseTemplates: {
+        'application/json': `{
+            "error": "Bad request!"
+          }`,
+      },
+    },
+    {
+      /**
+       * Match all '5XX' HTTP Errors
+       */
+      selectionPattern: '5\\d{2}',
+      statusCode: '500',
+      responseTemplates: {
+        'application/json': '"error": $input.path(\'$.error\')',
+      },
+    },
+  ];
+
+  const integResponse = [
+    {
+      statusCode: '200',
+      responseTemplates: {
+        /* eslint-disable */
+        'application/json': [
+          '#set($inputRoot = $input.path(\'$\'))',
+          '#if($input.path(\'$.status\').toString().equals("FAILED"))', 
+            '#set($context.responseOverride.status = 500)', 
+            '{',
+              '"error": "$input.path(\'$.error\')"',
+              '"cause": "$input.path(\'$.cause\')"',
+            '}',
+          '#else',
+            '$input.path(\'$.output\')',
+          '#end',
+        /* eslint-enable */
+        ].join('\n'),
+      },
+    },
+    ...errorResponse,
+  ];
+
+  return integResponse;
+}
+
+/**
+ * Defines the request template that will be used for the integration
+ * @param stateMachine
+ * @param options
+ * @returns requestTemplate
+ */
+function requestTemplates(stateMachine: sfn.IStateMachine, options: StepFunctionsExecutionIntegrationOptions) {
+  const templateStr = templateString(stateMachine, options);
+
+  const requestTemplate: { [contentType:string] : string } =
+    {
+      'application/json': templateStr,
+    };
+
+  return requestTemplate;
+}
+
+/**
+ * Reads the VTL template and returns the template string to be used
+ * for the request template.
+ *
+ * @param stateMachine
+ * @param includeRequestContext
+ * @param options
+ * @reutrns templateString
+ */
+function templateString(
+  stateMachine: sfn.IStateMachine,
+  options: StepFunctionsExecutionIntegrationOptions): string {
+  let templateStr: string;
+
+  let requestContextStr = '';
+
+  const includeHeader = options.headers?? false;
+  const includeQueryString = options.querystring?? true;
+  const includePath = options.path?? true;
```

Comment:
Need unit tests for this and request context. 
(For request context, just testing with one attribute is sufficient. No need to test for all attributes)

The assertion in the unit test will look something like 

```ts
expect(stack).toHaveResource('AWS::ApiGateway::Method', {
  RequestTemplates: {
    'application/json': {
      'Fn::Join': [
        '',
        [
          stringLike('*#set($includeHeaders = true)*'),
          { Ref: 'StateMachine2E01A3A5' },
          anything(),
        ],
      ],
    },
  },
});
```

Documentation for `stringLike()` and `anything()` can be found here - https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/assert-internal#check-existence-of-a-resource

Reply:
only:

```
expect(stack).toHaveResource('AWS::ApiGateway::Method', {
  RequestTemplates: {
    'application/json': {
   },
    ...
}
```

did not work. it said something like: "there are extra fields here", so I had to include the Integration. For the stringLike(), it did not work on multi-line strings.

I submitted a fix for that:

https://github.com/aws/aws-cdk/pull/17692/files

I had to commit the fixed file for the unit tests to succeed.